File: analytics/src/test/java/io/appmetrica/analytics/AdsIdentifiersResultTest.java
package io.appmetrica.analytics;

import io.appmetrica.analytics.assertions.ObjectPropertyAssertions;
import io.appmetrica.analytics.testutils.CommonTest;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static io.appmetrica.analytics.assertions.AssertionsKt.ObjectPropertyAssertions;
import static org.mockito.Mockito.mock;

@RunWith(RobolectricTestRunner.class)
public class AdsIdentifiersResultTest extends CommonTest {

    @Test
    public void testConstructor() throws Exception {
        AdsIdentifiersResult.AdvId gaid = mock(AdsIdentifiersResult.AdvId.class);
        AdsIdentifiersResult.AdvId hoaid = mock(AdsIdentifiersResult.AdvId.class);
        AdsIdentifiersResult.AdvId yandex = mock(AdsIdentifiersResult.AdvId.class);
        AdsIdentifiersResult result = new AdsIdentifiersResult(gaid, hoaid, yandex);
        ObjectPropertyAssertions<AdsIdentifiersResult> assertions = ObjectPropertyAssertions(result);
        assertions.checkField("googleAdvId", gaid);
        assertions.checkField("huaweiAdvId", hoaid);
        assertions.checkField("yandexAdvId", yandex);
        assertions.checkAll();
    }
}

File: analytics/src/test/java/io/appmetrica/analytics/AdsIdentifiersResultTest.kt
package io.appmetrica.analytics

import io.appmetrica.analytics.AdsIdentifiersResult.AdvId
import io.appmetrica.analytics.assertions.ObjectPropertyAssertions
import io.appmetrica.analytics.testutils.CommonTest
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.kotlin.mock
import org.robolectric.RobolectricTestRunner

@RunWith(RobolectricTestRunner::class)
class AdsIdentifiersResultTest : CommonTest() {

    @Test
    fun constructor() {
        val gaid = mock<AdvId>()
        val hoaid = mock<AdvId>()
        val yandex = mock<AdvId>()
        val result = AdsIdentifiersResult(gaid, hoaid, yandex)
        ObjectPropertyAssertions(result)
            .checkField("googleAdvId", gaid)
            .checkField("huaweiAdvId", hoaid)
            .checkField("yandexAdvId", yandex)
            .checkAll()
    }
}
